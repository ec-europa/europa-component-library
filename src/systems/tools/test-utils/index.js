const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
