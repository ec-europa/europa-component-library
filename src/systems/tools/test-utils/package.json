{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
