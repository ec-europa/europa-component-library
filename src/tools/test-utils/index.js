const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
