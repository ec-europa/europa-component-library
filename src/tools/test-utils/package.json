{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
