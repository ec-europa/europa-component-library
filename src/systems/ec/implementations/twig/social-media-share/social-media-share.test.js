import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

// Set fake paths for svgs to render for tests.
demoData.links.forEach((link) => {
  if (link.icon) {
    link.icon.forEach((icon) => {
      icon.path = 'example'; // eslint-disable-line no-param-reassign
    });
  }
});

describe('EC - Social Media Share', () => {
  const template =
    '@ecl-twig/ec-component-social-media-share/ecl-social-media-share.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);
  describe('Default', () => {
    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoData, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoData, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
  describe('Default view', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demoData)).resolves.toMatchSnapshot();
    });
  });
  describe('with validation enabled and missing input data', () => {
    test('returns the right warning message', () => {
      expect.assertions(1);

      const dataCompliance = {
        ...demoData,
        _compliance_: true,
      };
      dataCompliance.description = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });
});
