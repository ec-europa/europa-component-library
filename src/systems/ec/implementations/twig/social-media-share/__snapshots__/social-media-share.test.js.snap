// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Social Media Share Default renders correctly with extra attributes 1`] = `
<test>
  <div
    class="ecl-social-media-share"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <p
      class="ecl-social-media-share__description"
    >
      Share this page
    </p>
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--twitter"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#twitter_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--facebook"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#facebook_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--instagram"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#instagram_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--linkedin"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--email"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#email"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#email_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            E-mail
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-share__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</test>
`;

exports[`EC - Social Media Share Default renders correctly with extra class names 1`] = `
<test>
  <div
    class="ecl-social-media-share custom-class custom-class--test"
  >
    <p
      class="ecl-social-media-share__description"
    >
      Share this page
    </p>
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--twitter"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#twitter_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--facebook"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#facebook_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--instagram"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#instagram_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--linkedin"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--email"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#email"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#email_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            E-mail
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-share__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</test>
`;

exports[`EC - Social Media Share Default view renders correctly 1`] = `
<test>
  <div
    class="ecl-social-media-share"
  >
    <p
      class="ecl-social-media-share__description"
    >
      Share this page
    </p>
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--twitter"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#twitter"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#twitter_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--facebook"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#facebook"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#facebook_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--instagram"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#instagram"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#instagram_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--linkedin"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--email"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#email"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#email_hover"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            E-mail
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-share__link"
          href="/example"
        >
          Other social networks
        </a>
      </li>
    </ul>
  </div>
</test>
`;

exports[`EC - Social Media Share with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <div
    class="ecl-social-media-share"
  >
    <p
      class="ecl-social-media-share__description"
    />
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--twitter"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#twitter"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#twitter_hover"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
           
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
          >
            [link--standalone] - ecl compliant
          </p>
        </div>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--facebook"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#facebook"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#facebook_hover"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
           
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
          >
            [link--standalone] - ecl compliant
          </p>
        </div>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--instagram"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#instagram"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#instagram_hover"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
           
          <span
            class="ecl-link__label"
          >
            Instagram
          </span>
        </a>
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
          >
            [link--standalone] - ecl compliant
          </p>
        </div>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--linkedin"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#linkedin_hover"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
           
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
          >
            [link--standalone] - ecl compliant
          </p>
        </div>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link ecl-social-media-share__link--email"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="example#email"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xl ecl-link__icon ecl-social-media-share__icon ecl-social-media-share__icon-hover"
            focusable="false"
          >
            <use
              xlink:href="example#email_hover"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="example#notifications--success"
              />
            </svg>
          </span>
           
          <span
            class="ecl-link__label"
          >
            E-mail
          </span>
        </a>
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
          >
            [link--standalone] - ecl compliant
          </p>
        </div>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-social-media-share__link"
          href="/example"
        >
          Other social networks
        </a>
        <!-- ecl2-compliance: link--standalone -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
          >
            [link--standalone] - ecl compliant
          </p>
        </div>
      </li>
    </ul>
  </div>
  <!-- ecl2-compliance: social-media--share -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [social-media--share] -  {{ description }} needs to be set and filled in.
    </p>
  </div>
</test>
`;
